ui/app/models/pki/role.js

```diff
@@ -299,65 +296,62 @@ export default class PkiRoleModel extends Model {
   }
 
   get fieldGroups() {
-    if (!this._fieldToAttrsGroups) {
-      this._fieldToAttrsGroups = fieldToAttrs(this, [
-        {
-          default: [
-            'name',
-            'issuerRef',
-            'customTtl',
-            'notBeforeDuration',
-            'maxTtl',
-            'generateLease',
-            'noStore',
-            'addBasicConstraints',
-          ],
-        },
-        {
-          'Domain handling': [
-            'allowedDomains',
-            'allowedDomainsTemplate',
-            'allowBareDomains',
-            'allowSubdomains',
-            'allowGlobDomains',
-            'allowWildcardCertificates',
-            'allowLocalhost', // default: true (returned true by OpenApi)
-            'allowAnyName',
-            'enforceHostnames', // default: true (returned true by OpenApi)
-          ],
-        },
-        {
-          'Key parameters': ['keyType', 'keyBits', 'signatureBits'],
-        },
-        {
-          'Key usage': ['keyUsage', 'extKeyUsage', 'extKeyUsageOids'],
-        },
-        { 'Policy identifiers': ['policyIdentifiers'] },
-        {
-          'Subject Alternative Name (SAN) Options': [
-            'allowIpSans',
-            'allowedUriSans',
-            'allowUriSansTemplate',
-            'allowedOtherSans',
-          ],
-        },
-        {
-          'Additional subject fields': [
-            'allowedSerialNumbers',
-            'requireCn',
-            'useCsrCommonName',
-            'useCsrSans',
-            'ou',
-            'organization',
-            'country',
-            'locality',
-            'province',
-            'streetAddress',
-            'postalCode',
-          ],
-        },
-      ]);
-    }
-    return this._fieldToAttrsGroups;
+    return fieldToAttrs(this, [
```

Comment:
I think using `fieldToAttrs` directly caused some bugginess which is why we were setting it to a private variable. I chatted with @zofskeez to try and remember what exactly was happening - he thinks that each time the getter fired and ran `expandAttributeMeta` the value was getting set back to the default so for an input it seemed like the value wasn't getting set. (I'm having trouble remember the exact glitchy behavior, but it made form inputs un-inputtable) 

I think you could use the `@withFormFields()` decorator, though! 😄

Reply:
ah ok, thanks for the context, I didn't see why this was like this. I'll test out the decorator, thanks!